src/Microsoft.AspNetCore.OData/Batch/ODataBatchReaderExtensions.cs

```diff
@@ -227,6 +242,23 @@ private static HttpContext CreateHttpContext(HttpContext originalContext)
             context.Request.Cookies = originalContext.Request.Cookies;
             foreach (KeyValuePair<string, StringValues> header in originalContext.Request.Headers)
             {
+                var headerKey = header.Key.ToLowerInvariant();
+                // do not copy over headers that should not be inherited from batch to individual requests
+                if (headerKey == "content-length" ||
+                    headerKey == "content-type")
+                {
+                    continue;
+                }
+                // some preferences may be inherited, others discarded
+                if (headerKey == "prefer")
```

Comment:
Could invert the ```if``` to take care of the one liner and leave the other code indented less...